src/verifast1.ml

```diff
@@ -1823,6 +1838,18 @@ module VerifyProgram1(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
   let classterms = classterms1 @ classterms0
   let interfaceterms = interfaceterms1 @ interfaceterms0
 
+  let cxx_type_info_map1 =
+    structdeclmap |> List.fold_left begin fun acc (sn, (_, body_opt, _)) ->
+      match body_opt with
+      | None ->
+        acc
+      | Some _ ->
+        let type_info = get_unique_var_symb (sn ^ "_type_info") cxx_type_info_ref_type in

```

Comment:
The provertype of `RefType` is currently still `int`; it should be `inductive` to match the provertype of pointers (since references are really just pointers).

Reply:
Try running the C++ examples with Z3; it does typechecking of the terms.

Reply:
Changed the provertype of `RefType` to `inductive`